src/Data/Lazy.ss

```diff
@@ -0,0 +1,17 @@
+;; -*- mode: scheme -*-
+
+(library (Data.Lazy foreign)
+  (export defer
+          force
+  )
+
+  (import (only (rnrs base) define lambda))
+
+  (define defer
+    (lambda (thunk)
+      (lambda ()
+        (thunk 'unit))))
+
+  (define force
+    (lambda (thunk) (thunk)))
```

Comment:
I think you also need pass in the `'unit` here as well.

Reply:
Oh wait no this won't have that issue as it's the `Lazy` here that we construct in `defer`.